src/Motor.Extensions.Diagnostics.Metrics/AutoObserveStopwatch.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Prometheus.Client;
+
+namespace Motor.Extensions.Diagnostics.Metrics
+{
+    public class AutoObserveStopwatch : IDisposable
+    {
+        private readonly Stopwatch _watch;
+        private readonly ISummary? _unlabeledSummary;
+        private readonly IMetricFamily<ISummary>? _labeledSummary;
+        private readonly AbstractLabel[] _labels = Array.Empty<AbstractLabel>();
+
+        public AutoObserveStopwatch(ISummary? unlabeledSummary)
+        {
+            _unlabeledSummary = unlabeledSummary;
+            _watch = Stopwatch.StartNew();
+        }
+
+        public AutoObserveStopwatch(IMetricFamily<ISummary>? labeledSummary, params AbstractLabel[] labels)
```

Comment:
Any reason you don't support `IMetricsFamily<IMetric, TLabels>`? :sweat_smile:

Reply:
Can you provide me an example where this is used? I'm curious where the difference is between these two.